# This patch enables xscreensaver-getimage-file to work sequentially,
# allowing users to run a slideshow in order.
#
# To use in this mode, edit ~/.xscreensaver updating:
#    chooseRandomImages: True
#    imageDirectory: /path/to/ordered/pictures
#
# And then:
#    echo 0 > ~/.xscreensaver-image-index
#
# Then run:
#    /usr/lib/xscreensaver/glslideshow -duration 2 -pan 2
#
# To disable:
#    rm ~/.xscreensaver-image-index
#
# patch by: Dustin Kirkland <kirkland@ubuntu.com>

=== modified file 'driver/xscreensaver-getimage-file'
--- a/driver/xscreensaver-getimage-file	2009-09-23 20:47:27 +0000
+++ b/driver/xscreensaver-getimage-file	2010-11-25 22:25:50 +0000
@@ -359,7 +359,20 @@
   my $max_tries = 50;
   for (my $i = 0; $i < $max_tries; $i++) {
 
+    # Default to random image
     my $n = int (rand ($#all_files + 1));
+
+    # But check if the user wants a sequential image
+    if (open(IN, "<$ENV{HOME}/.xscreensaver-image-index")) {
+         read(IN, $n, 8);
+         close(IN);
+         $n++;
+         if (open (IN, ">$ENV{HOME}/.xscreensaver-image-index")) {
+             print(IN $n--);
+             close(IN);
+         }
+    }
+
     my $file = $all_files[$n];
     if (large_enough_p ($file)) {
       return $file;

